mlflow/artifacts/__init__.py

```diff
@@ -57,3 +62,108 @@ def download_artifacts(
     artifact_repo = get_artifact_repository(artifact_uri)
     artifact_location = artifact_repo.download_artifacts(artifact_path, dst_path=dst_path)
     return artifact_location
+
+
+def load_text(artifact_uri: str) -> str:
+    """
+    Loads content of the artifact file as python str object
+    :param artifact_uri: artifact location
+    :return: str
+
+    .. code-block:: python
+        :caption: Example
+
+        import mlflow
+        with mlflow.start_run() as run:
+            artifact_uri = run.info.artifact_uri
+            mlflow.log_text("This is a sentence", "file.txt")
+            file_content  = mlflow.artifacts.load_text(artifact_uri + "/file.txt")
+            print(file_content)
+
+    .. code-block:: text
+        :caption: Output
+
+        This is a sentence
+    """
+    local_artifact = download_artifacts(artifact_uri)
+    with open(local_artifact) as local_artifact_fd:
+        try:
+            return str(local_artifact_fd.read())
+        except Exception as e:
+            _logger.exception(e)
+            raise MlflowException("Unable to form a str object from file content", BAD_REQUEST)
```

Comment:
Do we need this try-catch?

Reply:
I thought it is good to have. What if user tries to use the `load_text` method with an artifact file that contains image ?

```
with mlflow.start_run() as run:
    image = Image.new("RGB", (100, 100))
    artifact_uri = run.info.artifact_uri
    mlflow.log_image(image, "image.png")
    artifact_uri = run.info.artifact_uri
    file_content = mlflow.artifacts.load_text(artifact_uri + "/image.png")
    print(file_content)
```

Error traceback
```
Traceback (most recent call last):
  File "/Users/subras/github/mlflow-fork/mlflow/artifacts/__init__.py", line 93, in load_text
    return str(local_artifact_fd.read())
  File "/Users/subras/.pyenv/versions/3.7.13/lib/python3.7/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x89 in position 0: invalid start byte

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 47, in <module>
    file_content = mlflow.artifacts.load_text(artifact_uri + "/image.png")
  File "/Users/subras/github/mlflow-fork/mlflow/artifacts/__init__.py", line 95, in load_text
    raise MlflowException("Unable to form a str object from file content", BAD_REQUEST)
mlflow.exceptions.MlflowException: Unable to form a str object from file content
```